.expeditor/scripts/release_habitat/package_and_upload_binary.sh

```diff
@@ -7,128 +7,86 @@ set -euo pipefail
   exit 1
 fi
-tail -n+6 "${target_hart}" | \
-    tar --directory "${extract_dir}" \
-        --extract \
+
+# NOTE: The -r for jq is VERY IMPORTANT (otherwise you get literal '"'
+# characters in the resulting string).
+hab_version=$(${hab_binary} pkg info --json "${hart}" | jq -r '.version')
+hab_release=$(${hab_binary} pkg info --json "${hart}" | jq -r '.release')
+
+# We'll extract our binaries into a directory with this name. This
+# will also be the sole directory within the resulting archive.
+#
+# e.g. "hab-0.88.0-20191009205851-x86_64-linux"
+archive_dir="hab-${hab_version}-${hab_release}-${BUILD_PKG_TARGET}"
+mkdir "${archive_dir}"
+
+# This bit of magic strips off the Habitat header (first 6 lines) from
+# the compressed tar file that is a core/hab .hart, and extracts the
+# contents of the `bin` directory only, into the ${archive_dir}
+# directory.
+#
+# For Linux and macOS packages, this will just include the single
+# `hab` binary, but on Windows, it will include `hab.exe`, as well as
+# all the DLL files needed to run it.
+#
+# At the end of the day, that's all we need to package up in a
+# Habitat-agnostic archive.
+tail --lines=+6 "${hart}" | \
+    tar --extract \
+        --directory="${archive_dir}" \
         --xz \
-        --strip-components=6
-
-extracted_hab_binary="$(find "$extract_dir" \( -name hab -or -name hab.exe \) -type f)"
-pkg_ident="$(tr --delete '\r' < "$extract_dir"/IDENT)"
-pkg_version="$(echo "$pkg_ident" | cut -d '/' -f 3)"
-pkg_release="$(echo "$pkg_ident" | cut -d '/' -f 4)"
-archive_name="hab-${pkg_version}-${pkg_release}-$BUILD_PKG_TARGET"
-
-build_dir="$tmp_root/build"
-pkg_dir="$build_dir/${archive_name}"
-
-echo "Copying $extracted_hab_binary to $(basename "$pkg_dir")"
-mkdir -p "$pkg_dir"
-mkdir -p "$tmp_root/results"
-
-if [[ $BUILD_PKG_TARGET == *"windows" ]]; then
-  # windows has more than just the hab binary, so we need to copy all the files
-  # in the directory
-  for file in "$(dirname "$extracted_hab_binary")"/*; do 
-    cp -p "$file" "$pkg_dir/"
-  done
-else
-  cp -p "$extracted_hab_binary" "$pkg_dir/$(basename "$extracted_hab_binary")"
-fi
+        --strip-components=7 \
+        --wildcards "hab/pkgs/core/hab/*/*/bin/"
+
+echo "--- Compressing 'hab' binary"
+
+pkg_name="hab-${BUILD_PKG_TARGET}"
 
-echo "Compressing 'hab' binary"
 case "$BUILD_PKG_TARGET" in
-*-linux | *-linux-kernel2)
-    pkg_artifact="$tmp_root/results/${archive_name}.tar.gz"
-    tarball="$(basename --suffix=".gz" "${pkg_artifact}")"
-    hab pkg exec core/tar tar cf "$tarball" "$build_dir/$(basename "$pkg_dir")"
-    hab pkg exec core/gzip gzip -9 -c "$tarball" > "$pkg_artifact"
-    ;;
-*-darwin | *-windows)
-    pkg_artifact="$tmp_root/results/${archive_name}.zip"
-    hab pkg exec core/zip zip -9 -r "$pkg_artifact" "$build_dir/$(basename "$pkg_dir")"
-    ;;
-*)
-    exit_with "$target_hart has unknown TARGET=$BUILD_PKG_TARGET" 3
-    ;;
+    *-linux | *-linux-kernel2)
```

Comment:
This looks right to me, but if we haven't already it'd be good to compare this output with existing releases.

Reply:
I compared it to the existing ones, and it looks equivalent.

Wouldn't be too tough to create a test for that, though... just gotta figure out where the best place to put such a thing would be :thinking: